blockchain/state_transition.go

```diff
@@ -126,13 +135,17 @@ type kerror struct {
 
 // NewStateTransition initialises and returns a new state transition object.
 func NewStateTransition(evm *vm.EVM, msg Message) *StateTransition {
+	effectiveGasPrice := msg.EffectiveGasPrice(evm.BaseFee)
```

Comment:
How about using `evm.GasPrice` instead of recalculating?

Reply:
In my opnion, it seems better to safely compute the `effectiveGasPrice` separately from the msg variable.
In the current code, gasPrice of EVM and effectiveGasPrice have the same value. But in the future, it may have different value each other.